plugins/3-pip.sh

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+test -f requirements.txt || return 1
+
+pip_bootstrap(){
+  run pip install -r requirements.txt
+}
```

Comment:
I feel like with installing pip you'd really want to only install requirements inside a [virtualenv](https://virtualenv.pypa.io/en/stable/) otherwise it'll be installed globally and things become get weird when you have multiple python projects.

Reply:
This could be `run python -m venv .venv && .venv/bin/pip install -r requirements.txt`, but that's making some assumptions about the project.

Reply:
Using one of the common virtual environment destinations in the default Python [gitignore](https://github.com/github/gitignore/blob/master/Python.gitignore#L88) wouldn't be the worst solution, imho. Better than installing to the system Python.

Reply:
Do most people use vitualenv with pip?

Reply:
Yes.

Nearly every CI and PaaS service creates a virtual environment to run tests and the server, respectively. Python 3.3+ [includes](https://docs.python.org/3/library/venv.html) virtual environment support (`python -m venv`), but any version of Python can use `virtualenv` to create virtual environments [if installed](https://virtualenv.pypa.io).

Inside the created virtual environment, `pip` is used to install dependencies.

Reply:
> Better than installing to the system Python.

I agree. Creating the virtual environment at `./.venv/` is the least astonishing location.

Reply:
And I think it's safe assume the project is running on at least Python 3.3.